File: src/main/java/me/racci/sylphia/events/DataLoadEvent.java
package me.racci.sylphia.events;

import me.racci.sylphia.data.PlayerData;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

public class DataLoadEvent extends Event {

    private static final HandlerList handlers = new HandlerList();

    private final PlayerData playerData;

    public DataLoadEvent(PlayerData playerData) {
        this.playerData = playerData;
    }

    public PlayerData getPlayerData() {
        return playerData;
    }

    @NotNull
    @Override
    public HandlerList getHandlers() {
        return handlers;
    }
}

File: src/main/kotlin/me/racci/sylphia/events/DataLoadEvent.kt
@file:Suppress("unused")
@file:JvmName("DataLoadEvent")
package me.racci.sylphia.events

import me.racci.sylphia.data.PlayerData
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class DataLoadEvent(val playerData: PlayerData) : Event(true) {
    override fun getHandlers(): HandlerList {
        return Companion.handlers
    }

    companion object {
        private val handlers = HandlerList()
    }
}